### PR TITLE
Comments Implementation

### PR DESCRIPTION
**Pull request by RakeshMaddala**
Comments can be added against the unapproved challenges as part of review content

### CODE REVIEW

File: ideaforce/src/main/java/com/emc/ideaforce/controller/AdminController.java [hunk 1]
```diff
@@ -44,9 +53,33 @@ public ModelAndView showAdminPage(Principal principal) {
...
+    @RequestMapping(value="/addcomments", method = POST)
+    public ModelAndView addCommentsForStory(@ModelAttribute("newcomment") CommentDto commentModel,
+                                            Principal principal) {
+        User currentUser = userService.getUser(principal.getName());
+        commonService.saveStoryComment(commentModel, currentUser);
+        return showAdminPage();
+    }
+
+    @RequestMapping(value="/viewcomments/{id}", method = GET)
+    public ModelAndView viewCommentsForStory(@PathVariable String id) {
```
Comment (bharathiraja-subramanian) [inline ideaforce/src/main/java/com/emc/ideaforce/controller/AdminController.java]: This api is also required for for normal_user, right? User needs to view the comments given for his story..

File: ideaforce/src/main/java/com/emc/ideaforce/controller/AdminController.java [hunk 2]
```diff
@@ -44,9 +53,33 @@ public ModelAndView showAdminPage(Principal principal) {
...
+    @RequestMapping(value="/addcomments/{id}", method = GET)
+    public ModelAndView commentsForStory(@PathVariable String id, ModelMap model) {
+        CommentDto commentDto = new CommentDto();
+        commentDto.setStoryId(id);
+        model.addAttribute("newcomment", commentDto);
+        return new ModelAndView(ADD_COMMENTS_VIEW, model);
+    }
+
+    @RequestMapping(value="/addcomments", method = POST)
+    public ModelAndView addCommentsForStory(@ModelAttribute("newcomment") CommentDto commentModel,
```
Comment (bharathiraja-subramanian) [inline ideaforce/src/main/java/com/emc/ideaforce/controller/AdminController.java]: We do not need these many apis, following would be ssufficient
`/viewcomments/{id}` -> `/review/{id}` (GET)
`/addcomments` -> `/review/{id}` (POST)
`/addcomments/{id}` -> `/review/{id}` (POST)
`/approve/{id}` -> `/review/{id}` (POST)

Restructuring `CommentDto` to `ReviewDto` and have the `reviewstatus` and `comments` inside that obj

File: ideaforce/src/main/java/com/emc/ideaforce/controller/AdminController.java (hunk 1 above)
Comment (RakeshMaddala) [reply ideaforce/src/main/java/com/emc/ideaforce/controller/AdminController.java]: Can it be part of profile implementation ?

Comment (quailbr) [reply ideaforce/src/main/java/com/emc/ideaforce/controller/AdminController.java]: Sure

File: ideaforce/src/main/java/com/emc/ideaforce/controller/AdminController.java (hunk 2 above)
Comment (RakeshMaddala) [reply ideaforce/src/main/java/com/emc/ideaforce/controller/AdminController.java]: I am not able structure apis in this format. Need ```/addcomments/{id}``` for routing ui to comment page.

**Comment by quailbr** [comment]
@RakeshMaddala ,
As discussed, merging it now but we will have to do some refactoring on this

**Pull request merged**